examples/NDBC_data.ipynb

```diff
@@ -0,0 +1,194 @@
+    "plot_map(east_coast_stations, 'NDBC Stations on the East Coast of the US')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Not all stations provide the same set of measurements, the following measurements are listed and which type of stations support them:\n",
+    "\n",
+    "```markdown\n",
+    "\n",
+    "| Package Name          | C-MAN  | Moored |\n",
+    "|-----------------------|--------|--------|\n",
+    "| adcp                  | [ ]    | [ ]    |\n",
+    "| cwind                 | [X]    | [ ]    |\n",
+    "| ocean                 | [ ]    | [ ]    |\n",
+    "| spec                  | [ ]    | [ ]    |\n",
+    "| stdmet                | [X]    | [ ]    |\n",
+    "| supl                  | [ ]    | [X]    |\n",
+    "| swden                 | [ ]    | [X]    |\n",
+    "| swdir                 | [ ]    | [X]    |\n",
+    "| swdir2                | [ ]    | [X]    |\n",
+    "| swr1                  | [ ]    | [X]    |\n",
+    "| swr2                  | [ ]    | [X]    |\n",
+    "```\n",
+    "\n",
+    "\n",
+    "(note: stations that are with only numbers are usually Moored Buoyse e.g. 44008 and stations with a mixture of numbers and letters are C-MAN owned e.g. BUZM3)\n",
+    "C-MAN -> Numbers only\n",
+    "Moored Buoyse -> numbers and letters\n",
+    "\n",
+    "The table was tested at the start of june 2024 and may differ as support for different stations status may change.\n",
+    "\n",
+    "Note: If an error is logged,it may be because one of the stations does not have the data of the mode\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Retrieve Data for Selected Stations"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define start and end dates for data retrieval\n",
+    "start_date = \"2023-01-01\"\n",
+    "end_date = \"2023-01-10\"\n",
+    "\n",
+    "# Get data for selected stations (mode = 'stdmet' for standard meteorological data)\n",
+    "station_ids = east_coast_stations[\"Station\"].tolist()[1:3]\n",
+    "data_df = fetch_ndbc_station(\n",
+    "    station_ids=[\"SRST2\",\"VENF1\"],\n",
+    "    mode=\"stdmet\",\n",
+    "    start_date=start_date,\n",
+    "    end_date=end_date,\n",
+    ")\n",
+    "\n",
+    "#output is a dictionary of {station id : data}\n",
+    "data_to_plot = data_df[\"SRST2\"]\n",
+    "data_to_plot"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Plot the data of the dataframes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot the data of data_to_plot\n",
+    "data_to_plot[[\"WSPD\", \"GST\", \"ATMP\"]].rename(columns={\"WSPD\": \"Wind Speed\", \"GST\": \"Wind Gust\", \"ATMP\": \"Air Temperature\"}).hvplot(grid=True)"
```

Comment:
Practically, since we implement a new provider that provider mainly wave, I'd like to see some visualizations of wave data. 
As I am writing this review, Beryl is ravaging the Caribbean and the 3 following station are on its path: 
"42056", "42058", "42059"

But more generally, I'd like to see: 
 * one plot for discrete spectral parameters along the time
 * one spectral plot for one time step (see how it's done here in [wavespectra](https://github.com/wavespectra/wavespectra/blob/master/wavespectra/plot.py)), just a simple polar plot would be enough, to avoid pulling that package's dependencies
 * one plot of temporal variation of spectral density (like [here](https://www.researchgate.net/publication/361420252_Characteristics_of_Horizontal_to_Vertical_Spectral_Ratio_of_InSight_Seismic_Data_From_Mars/figures)).

Reply:
> one spectral plot for one time step (see how it's done here in [wavespectra](https://github.com/wavespectra/wavespectra/blob/master/wavespectra/plot.py)), just a simple polar plot would be enough, to avoid pulling that package's dependencies

Forget it, I found out that spectro-angular density data is not delivered by NDBC stations (could you confirm this @AliS-Noaa ?). It is only along the frequency bands, not the directions. 

I was thinking about this: [wavewatch spectrum](https://polar.ncep.noaa.gov/waves/pres/primer/primer_11.html).

Reply:
> 1. one plot for discrete spectral parameters along the time
Something like: 
```python
# Define start and end dates for data retrieval
start_date = "2023-06-26"
end_date = "2024-07-03"

# Get data for selected stations (mode = 'stdmet' for standard meteorological data)
data_df = fetch_ndbc_station(
    station_ids=["42056", "42058", "42059"],
    mode="stdmet",
    start_date=start_date,
    end_date=end_date,
)

#output is a dictionary of {station id : data}
data_to_plot = data_df["42059"]
data_to_plot[["WVHT","WSPD"]].dropna().hvplot.line(xlabel='Time', title='Swell Wave Height')
```
![image](https://github.com/oceanmodeling/searvey/assets/18373442/14e8bc5f-b112-4dcd-b89d-b06b6a635bb0)

> 3. one plot of temporal variation of spectral density (like [here](https://www.researchgate.net/publication/361420252_Characteristics_of_Horizontal_to_Vertical_Spectral_Ratio_of_InSight_Seismic_Data_From_Mars/figures)).

Something like: 

```python
# Define start and end dates for data retrieval
start_date = "2024-05-01"
end_date = "2024-07-04"

# Get data for selected stations (mode = 'stdmet' for standard meteorological data)
data_df = fetch_ndbc_station(
    station_ids=["42059"],
    mode="swden",
    start_date=start_date,
    end_date=end_date,
)

#output is a dictionary of {station id : data}
data_to_plot = data_df["42059"]
data_to_plot.T.hvplot.heatmap(title = 'Wave Density evolution', cmap = 'rainbow4')
```
![image](https://github.com/oceanmodeling/searvey/assets/18373442/42721754-d967-499a-afdc-bc40999e86a6)

Also: I think that the `swden` take time to be uploaded. The whole month of June is empty.

Reply:
Do you want to commit this to the example usage file or should I? Because this is your work, and its really good.

Reply:
@abdu558 please go ahead and add this to the example notebook. Thanks!